### DISCUSSION TITLE
Show potential exceptions raised by functions

### DISCUSSION THREAD

**Discussion by tdkbzh** [microsoft/pylance-release#3351] [Enhancement]
When writing Python code, we do not know which exceptions can be raised by a function:

```python
import somelib

somelib.function() # which exceptions can be raised?
```

It would be nice to indicate to the developer which exceptions can be raised by a function call (or by imports).

**Comment by bschnurr** [comment]
thanks for the suggestion. 
We could add info to the signature docstring when hovering.

**Comment by tdkbzh** [comment]
Thanks for responding ;

I think one difficulty is that the function can call dependencies, which themselves can raise.

```python
from random import random

class CustomException1(Exception):
    pass

class CustomException2(Exception):
    pass

def dependency():
    if random() > .5:
        raise CustomException1()

def function():
    dependency()
    raise CustomException2()


function() # raises either CustomException1 or CustomException2
```
